crates/ruff_db/src/files.rs

```diff
@@ -20,16 +21,23 @@ mod path;
 
 /// Interns a file system path and returns a salsa `File` ingredient.
 ///
-/// Returns `None` if the path doesn't exist, isn't accessible, or if the path points to a directory.
+/// Returns `Err` if the path doesn't exist, isn't accessible, or if the path points to a directory.
 #[inline]
-pub fn system_path_to_file(db: &dyn Db, path: impl AsRef<SystemPath>) -> Option<File> {
+pub fn system_path_to_file(
+    db: &dyn Db,
+    path: impl AsRef<SystemPath>,
+) -> Result<File, SystemPathError> {
```

Comment:
Should `vendored_path_to_file()` also return a `Result`, for symmetry? Unlike `system_path_to_file`, we don't have a need right now to distinguish directories from files for vendored files... but the asymmetry between the two functions is a _little_ odd now, IMO

Reply:
I take another look tomorrow. Overall it feels overkill to return the same type only to get symmetry. Especially because it may also require quiet a bit of boilerplate (or we return more errors than are possible).